### PR TITLE
add api service

### PR DESCRIPTION
**Pull request by professahhh**
This PR adds GraphQL API integration.

### CODE REVIEW

File: src/enums/entity-status.enum.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const EntityStatus = {
```
Comment (professahhh) [inline src/enums/entity-status.enum.js:1]: I think I saw something else representing this so we may be able to remove this

File: src/graphql/queries.js [hunk 2]
```diff
@@ -0,0 +1,513 @@
+import { getCurrentDate } from '../utils/date';
+
+export const getRosterById = /* GraphQL */ `
```
Comment (professahhh) [inline src/graphql/queries.js:3]: Custom created for loading roster by site ID

File: src/graphql/queries.js [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const getSitesByProgramManager = /* GraphQL */ `
```
Comment (professahhh) [inline src/graphql/queries.js:38]: Custom created for getting program manager information, including their viable sites.

File: src/services/api.service.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            return youthWrapper.youth;
+        }).filter((youth) => youth.status === EntityStatus.Active),
+        siteAddress: result.data.getSite.address,
+        siteAdminEmail: result.data.getSite.siteAdminEmail,
+        siteAdminName: result.data.getSite.siteAdminName,
+        siteName: result.data.getSite.name,
+    };
+};
+
+export const getProgramManager = async (email) => {
```
Comment (professahhh) [inline src/services/api.service.js:29]: It's assumed the program manager email is available upon logging in. That should be passed into this query to fetch the manager's data including the sites they manage. The selected site ID can then be passed into the function above (`getSite`) to fetch the roster

**Review by jprejean-pariveda — APPROVED** [review]

**Pull request merged**